### DISCUSSION TITLE
`#[oai(extract = true)]` got error

### DISCUSSION THREAD

**Discussion by pymongo** [poem-web/poem#40] [Q&A] [answered]
I want to migrate my poem route handler to poem-openapi

I read the document here, but when I add `#[oai(extract = true)]` got error:

https://github.com/poem-web/poem/blob/d53c91c339dae9a1ea18845bbfb8778001a7892b/poem-openapi/src/docs/openapi.md#L18

![Screenshot_20210930_211239](https://user-images.githubusercontent.com/42347370/135461741-c6031f61-c6bf-4887-b758-f8a53054260e.png)

Here is the StateExtractor define, it works in normal poem handler function

```rust
pub struct StateExtractor {
    pub cookie_jar: CookieJar,
    pub graph_service: Arc<GraphService>,
    pub session_store: MemoryStore,
}

#[tonic::async_trait]
impl<'a> FromRequest<'a> for &'a StateExtractor {
    type Error = std::convert::Infallible;

    async fn from_request(req: &'a Request, _body: &mut RequestBody) -> Result<Self, Self::Error> {
        Ok(req.extensions().get::<StateExtractor>().unwrap())
    }
}
```

**Comment by sunli829** [comment, marked as answer]
Thank you for reporting this, I fixed it and added the test in `v0.7.7`. 🙂

https://github.com/poem-web/poem/blob/421e858a8bdc64fec24515793ddbc45308a64cdd/poem-openapi/tests/api.rs#L372
